library/c/sandesh.h

```diff
@@ -25,6 +25,52 @@ extern "C" {
 #include <Wsk.h>
 #include <Ws2ipdef.h>
 
+static void *
+s_malloc(unsigned int size)
+{
+    void *mem = ExAllocatePoolWithTag(NonPagedPool, size+1, 'RVCO');
+    *(int*)mem = size;
+    mem = (char*)mem + sizeof(unsigned int);
+    return mem;
+}
+
+static void *
+s_zalloc(unsigned int size)
+{
+    void *mem = s_malloc(size);
+    NdisZeroMemory(mem, size);
```

Comment:
According to MSDN `RtlZeroMemory` is preferred (see **Requirements** in https://msdn.microsoft.com/en-us/library/windows/hardware/ff564698(v=vs.85).aspx)